joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/RubySrc2Cpg.scala

```diff
@@ -1,26 +1,51 @@
 package io.joern.rubysrc2cpg
 
-import io.joern.rubysrc2cpg.passes.{AstCreationPass, ConfigPass}
+import better.files.File
+import io.joern.rubysrc2cpg.passes.{AstCreationPass, AstPackagePass, ConfigPass}
+import io.joern.rubysrc2cpg.utils.PackageTable
 import io.joern.x2cpg.X2Cpg.withNewEmptyCpg
 import io.joern.x2cpg.X2CpgFrontend
+import io.joern.x2cpg.datastructures.Global
 import io.joern.x2cpg.passes.frontend.{MetaDataPass, TypeNodePass}
 import io.shiftleft.codepropertygraph.Cpg
 import io.shiftleft.codepropertygraph.generated.Languages
 import org.slf4j.LoggerFactory
 
-import scala.util.Try
+import scala.sys.process._
+import scala.util.{Failure, Success, Try}
 
 class RubySrc2Cpg extends X2CpgFrontend[Config] {
 
   private val logger = LoggerFactory.getLogger(this.getClass)
 
+  val global = new Global()
+
   override def createCpg(config: Config): Try[Cpg] = {
     withNewEmptyCpg(config.outputPath, config: Config) { (cpg, config) =>
+      val packageTableInfo = new PackageTable()
       new MetaDataPass(cpg, Languages.RUBYSRC, config.inputPath).createAndApply()
       new ConfigPass(cpg, config.inputPath).createAndApply()
-      val astCreationPass = new AstCreationPass(config.inputPath, cpg)
+      if (config.enableDependencyDownload) {
+        val tempDir = File.newTemporaryDirectory()
+        downloadDependency(config.inputPath, tempDir.toString())
+        new AstPackagePass(cpg, tempDir.toString(), global, packageTableInfo, config.inputPath).createAndApply()
+        tempDir.delete()
+      }
+      val astCreationPass = new AstCreationPass(config.inputPath, cpg, global, packageTableInfo)
       astCreationPass.createAndApply()
       new TypeNodePass(astCreationPass.allUsedTypes(), cpg).createAndApply()
     }
   }
+
+  private def downloadDependency(inputPath: String, tempDir: String): Unit = {
+    if (File(s"$inputPath/Gemfile").exists) {
```

Comment:
Use OS-agnostic notation, you can do this as follows:
```suggestion
    if (File(inputPath) / "Gemfile").exists) {
```